### PR TITLE
UDP + ICMP + ARP demo

### PR DESCRIPTION
**Pull request by Akribes**
You can now run

```sh
sudo ip netns exec colorlight ping 192.168.1.123
```
and the FPGA will respond to pings.

Borrows a commit from #73.

### CODE REVIEW

**Comment by Akribes** [comment]
Now also includes UDP!

**Comment by t-wallet** [comment]
Before I give a review, can you rebase on develop and make the tests pass? Just so I don't review stuff that already has been done

File: src/Clash/Cores/Ethernet/Examples/EchoStack.hs [hunk 1]
```diff
@@ -54,3 +66,65 @@ ipEchoStackC rxClk rxRst rxEn txClk txRst txEn mac ip = ckt
             |> packetBufferC d10 d4
             |> mapMeta swapIp
             |> ipTxStack @4 txClk txRst txEn mac
+fullStackC
```
Comment (t-wallet) [inline src/Clash/Cores/Ethernet/Examples/EchoStack.hs:71]: Missing haddock

File: src/Clash/Cores/Ethernet/Examples/EchoStack.hs [hunk 2]
```diff
@@ -54,3 +66,65 @@ ipEchoStackC rxClk rxRst rxEn txClk txRst txEn mac ip = ckt
...
+  macRxStack @4 rxClk rxRst rxEn mac
+  |> arpIcmpUdpStackC mac ip (mapMeta $ B.second swapPorts)
+  |> macTxStack txClk txRst txEn
+  where
+    swapPorts hdr@UdpHeaderLite{..} = hdr
+                                        { _udplSrcPort = _udplDstPort
+                                        , _udplDstPort = _udplSrcPort
+                                        }
+
+arpIcmpUdpStackC
```
Comment (t-wallet) [inline src/Clash/Cores/Ethernet/Examples/EchoStack.hs:107]: Missing haddock

**Review by t-wallet — APPROVED** [review]
Looks good! Just some missing documentation for exported components.

**Review by rowanG077 — APPROVED** [review]
Once haddock is added I'm good with this

**Pull request merged**